app/controllers/provider_interface/provider_interface_controller.rb

```diff
@@ -48,6 +49,27 @@ def check_data_sharing_agreements
       end
     end
 
+    def check_provider_relationship_permissions
+      return unless FeatureFlag.active?('enforce_provider_to_provider_permissions')
+      return unless current_provider_user
+      return if performing_provider_organisation_setup?
+
+      if current_provider_user.can_manage_organisations? && provider_relationship_permissions_need_setup?
```

Comment:
`can_manage_organisations?` in its current form is not specific enough. If I am a provider user with 'manage organisations' for provider A but without 'manage organisations' for provider B, I'll still be able to set up org permissions for provider B. Maybe something like `can_manage_organisation?(provider)`?

Reply:
This is now https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/2231/commits/136e838c36d3805972582f3e73c6b4f6d08f4815#diff-4402a7a0a04b6d6e60e6fb0da05a3a8fR52 a user-level permissions check.

Reply:
I think it's still the case that if I'm associated with A and B, A is set up but B isn't, and I can only manage organisations for A, this returns true even though I can't manage B's relationships?

Reply:
@duncanjbrown https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/2231/files#diff-be572e8dffe260e5ccdf6095c4e62b25R43 adds an organisation specific check.
So we have 
`ProviderUser#can_manage_organisaions?` - A general check used for displaying navigation items etc.
`ProviderAuthorization#can_manage_organisation?(provider:)` A check that the acting user can manage a specific provider organisation.

Reply:
I think this would still redirect the user if they had manage_orgs but one of the orgs they couldn't manage was a ratifying provider?

Reply:
Ah yes this needs to use `ProviderAuthorisation#can_manage_organisation?(provider:)` thanks!

Reply:
Ah yes I think the best way to mitigate this is to find the first `TrainingProviderPermission` with a training provider who is one of the current user's providers and `setup_at` is nil. Then pass this provider to `ProviderAuthorisation#can_manage_organisation?` which will check the user has permission specific to that org.

Reply:
@duncanjbrown @mnacos I've amended the `ProviderInterfaceController` filter here https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/2231/files#diff-9817fde727b731ffab81417f046ad3a9R70-R91 to check for training provider permissions needing setup and then to check that the training provider on this org-level permission is a provider where the user has user-level permissions to manage organisations. Hopefully this tightens up the check properly.

Reply:
Thanks - looks good to me!